consensus/src/util/time_service.rs

```diff
@@ -148,53 +148,43 @@ pub enum WaitingError {
     },
 }
 
-/// Attempt to wait until the current time exceeds the min_duration_since_epoch if possible
+/// Attempt to wait until the current time exceeds the timestamp_then if possible
 ///
-/// If the waiting time exceeds max_instant then fail immediately.
+/// If the waiting time exceeds the deadline, then fail immediately.
 /// There are 4 potential outcomes, 2 successful and 2 errors, each represented by
 /// WaitingSuccess and WaitingError.
 pub async fn wait_if_possible(
     time_service: &dyn TimeService,
-    min_duration_since_epoch: Duration,
-    max_instant: Instant,
+    timestamp_then: Duration,
+    deadline: Instant,
 ) -> Result<WaitingSuccess, WaitingError> {
-    // Fail early if waiting for min_duration_since_epoch would exceed max_instant
-    // Ideally, comparing min_duration_since_epoch and max_instant would be straightforward, but
-    // min_duration_since_epoch is relative to UNIX_EPOCH and Instant is not comparable.  Therefore,
-    // we use relative differences to do the comparison.
-    let current_instant = Instant::now();
-    let current_duration_since_epoch = time_service.get_current_timestamp();
-    if current_instant <= max_instant {
-        let duration_to_max_time = max_instant.duration_since(current_instant);
-        if current_duration_since_epoch <= min_duration_since_epoch {
-            let duration_to_min_time = min_duration_since_epoch - current_duration_since_epoch;
-            if duration_to_max_time < duration_to_min_time {
-                return Err(WaitingError::MaxWaitExceeded);
-            }
-        }
+    // are we already passed the deadline?
+    let instant_now = Instant::now();
+    if instant_now >= deadline {
+        return Err(WaitingError::WaitFailed {
+            current_duration_since_epoch: 0,
```

Comment:
well somewhat redundant, shouldn't this always be equal to tim_service.get_current_timestamp()?